### PR TITLE
[T5A2][T09-A3] Tang Di Feng

### PR DESCRIPTION
**Pull request by e0011840**

### CODE REVIEW

File: src/seedu/addressbook/logic/Logic.java [hunk 1]
```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if(command.isMutating()) {
```
Comment (okkhoy) [inline src/seedu/addressbook/logic/Logic.java:88]: you need to update the documentation and the tests to reflect the changed behavior of this method

File: src/seedu/addressbook/commands/Command.java [hunk 2]
```diff
@@ -39,9 +39,14 @@ public static String getMessageForPersonListShownSummary(List<? extends ReadOnly
     /**
      * Executes the command and returns the result.
      */
-    public abstract CommandResult execute();
+    public abstract CommandResult execute() throws Exception;
 
     /**
+     * returns true if data is manipulated
```
Comment (okkhoy) [inline src/seedu/addressbook/commands/Command.java:45]: the command can *potentially* mutate data. may be you can update the comment to reflect that.
e.g. add command doesn't mutate data if the user tries to add a duplicate entry.

**Review by okkhoy — COMMENTED** [review]
Notice that your tests fail! this is because, you haven't updated the tests to check for the changed behavior of execute command.

**Comment by okkhoy** [comment]
@e0011840 
Some comments added. Please ack & close the PR after reading comments.

**Comment by e0011840** [comment]
Noted with thanks!

**Pull request closed without merging**